flocker/node/agents/ebs.py

```diff
@@ -1245,6 +1254,11 @@ def detach_volume(self, blockdevice_id):
         if ebs_volume.state != VolumeStates.IN_USE.value:
             raise UnattachedVolume(blockdevice_id)
 
+        if ebs_volume.attachments:
+            attachment_state = ebs_volume.attachments[0]['State']
```

Comment:
For Flocker created volumes, there will only ever be at most one attachment entry, unless something's gone horribly wrong somewhere. That we have to check `volume.attachments[0]` is a quirk of using Boto3; we didn't have a list prior to that upgrade.